tests/integration/models/torch/lighting_modules/test_segmentation.py

```diff
@@ -0,0 +1,178 @@
+            predictions.append(preds)
+    return torch.cat(predictions, dim=0)
+
+
+@pytest.fixture
+def segmentation_module() -> LightningModule:
+    cfg = DictConfig(
+        {
+            "models": fixt_models.deeplabv3_plus_w_target,
+            "trainer": fixt_trainers.trainer_cfg_w_cpu_devices,
+            "losses": fixt_losses.jaccard_loss_w_target,
+        }
+    )
+    model = get_model(cfg.models, cfg.trainer)
+    losses = get_losses(cfg, "image-segmentation", Frameworks.torch)
+    optimizer_cfg = DictConfig(fixt_optimizers.adam_optim_w_target)
+
+    module = SemanticSegmentationLightningModule(
+        model=model, losses=losses, optimizer_cfg=optimizer_cfg
+    )
+
+    return module
+
+
+def test_training_with_checkpoint(segmentation_module: LightningModule):
+    checkpoint_dir = "checkpoints"
+    os.makedirs(checkpoint_dir, exist_ok=True)
+
+    checkpoint_callback = ModelCheckpoint(
+        dirpath=checkpoint_dir,
+        filename="segmentation_module-{epoch:02d}",
+        save_top_k=1,
+        monitor="loss",
+        mode="min",
+        save_weights_only=True,
+    )
+
+    trainer = Trainer(max_epochs=1, devices=1, callbacks=[checkpoint_callback])
+    dataset = DummyDataset(num_samples=100)
+    dataloader = DataLoader(dataset, batch_size=4)
+    trainer.fit(segmentation_module, train_dataloaders=dataloader)
+
+
+def test_training_without_checkpoint(segmentation_module: LightningModule):
+    trainer = Trainer(max_epochs=1, devices=1)
+    dataset = DummyDataset(num_samples=100)
+    dataloader = DataLoader(dataset, batch_size=4)
+    trainer.fit(segmentation_module, train_dataloaders=dataloader)
+
+
+def test_testing_without_checkpoint(segmentation_module: LightningModule):
+    trainer = Trainer(max_epochs=1, devices=1)
+    dataset = DummyDataset(num_samples=100)
+    dataloader = DataLoader(dataset, batch_size=4)
+    trainer.test(segmentation_module, dataloaders=dataloader)
+
+
+def test_predicting_without_checkpoint(segmentation_module: LightningModule):
+    trainer = Trainer(max_epochs=1, devices=1)
+    predict_dataloader = DataLoader(DummyDataset(10), batch_size=4)
+    predictions = predict(segmentation_module, predict_dataloader)
+
+
+## working ##
+def test_testing_with_checkpoint(segmentation_module: LightningModule):
+    checkpoint_path = "checkpoints"
+    os.makedirs(checkpoint_path, exist_ok=True)
+
+    # # Set up the trainer
+    # checkpoint_callback = ModelCheckpoint(dirpath=checkpoint_path, save_top_k=-1)
+    trainer = Trainer(
+        max_epochs=2, devices=1, default_root_dir=checkpoint_path
+    )
+
+    dataset = DummyDataset(num_samples=100)
+    dataloader = DataLoader(dataset, batch_size=4)
+
+    # Fit the model
+    trainer.fit(segmentation_module, dataloader)
+
+    # Test the loaded model
+    trainer.test(
+        segmentation_module,
+        ckpt_path="checkpoints/lightning_logs/version_0/checkpoints/epoch=1-step=50.ckpt",
```

Comment:
I believe checkpoint file is only in your local machine. You can either mock checkpoint usage or upload a file with checkpoint, or come up with other idea.

Reply:
fixed